alicat/serial.py

```diff
@@ -132,21 +132,175 @@ def set_gas(self, gas, retries=2):
         """Set the gas type.
 
         Args:
-            gas: The gas type, as a string. Supported gas types are:
+            gas: The gas type, as a string or integer. Supported gas types by string are:
                 'Air', 'Ar', 'CH4', 'CO', 'CO2', 'C2H6', 'H2', 'He', 'N2',
                 'N2O', 'Ne', 'O2', 'C3H8', 'n-C4H10', 'C2H2', 'C2H4',
                 'i-C2H10', 'Kr', 'Xe', 'SF6', 'C-25', 'C-10', 'C-8', 'C-2',
                 'C-75', 'A-75', 'A-25', 'A1025', 'Star29', 'P-5'
+
+                For the full gas table, please see page 52 of the controller manual here:
+                https://documents.alicat.com/manuals/DOC-MANUAL-MC.pdf
+
+                Gas mixes may only be called by their mix number.
         """
         self._test_controller_open()
 
-        if gas not in self.gases:
-            raise ValueError("{} not supported!".format(gas))
-        command = '{addr}$${gas}\r'.format(addr=self.address,
-                                           gas=self.gases.index(gas))
+        if isinstance(gas, str) is False:
+            num = True
+        else:
+            num = False
+
+        if num is True:
+            command = '{addr}$${index}\r'.format(addr=self.address,
+                                               index=gas)
+            self._write_and_read(command, retries)
+        else:
+            if gas not in self.gases:
+                raise ValueError("{} not supported!".format(gas))
+            command = '{addr}$${gas}\r'.format(addr=self.address, gas=self.gases.index(gas))
+            self._write_and_read(command, retries)
+
+        read_reg46 = self._write_and_read('{addr}$$R46\r'.format(addr=self.address), retries)
+        reg46 = int(read_reg46.split()[-1])
+        bits = [32768, 16384, 8192, 4096, 2048, 1024, 512]
+
+        for u in range(0, 7):
+            reg46 = reg46 - bits[u]
+            if reg46 < 0:
+                reg46 = reg46 + bits[u]
+            elif reg46 < 255:
+                break
+
+        if num is True:
+            if gas != reg46:
+                raise IOError("Cannot set gas, gas set to Air.")
+        else:
+            if self.gases.index(gas) != reg46:
+                raise IOError("Cannot set gas.")
+
+    def create_mix(self, mix_no, name, gas1, percent1, gas2, percent2, gas3=None, percent3=None, gas4=None, percent4=None, gas5=None, percent5=None, retries=2):
```

Comment:
You could probably clean up this function by thinking through some other argument patterns. Python allows some fun interactions here. Let's try a few.

As lists:
```python
create_mix(100, 'air', ['N2', 'O2', 'Ar'], [78, 21, 1])
```
You end up making the lists below so this cleans up your function. But it's kind of awkward looking to use.

What about a dictionary:

```python
def create_mix(self, mix_no, name, gases):
    pass
create_mix(100, 'air', {'N2': 78, 'O2': 21, 'Ar': 1})
```
I really like this one. It'll allow you to really shorten your error checking code.

One more for fun would be kwarg unpacking.

```python
def create_mix(self, mix_no, name, **gases):
    pass
create_mix(100, 'air', N2=78, O2=21, Ar=1)
```
That's nice too and would have the same benefits as passing the dictionary above. The main downside here is that it's a more complex python idiom which can slow new coders down.